op-chain-ops/cmd/celo-migrate/main.go

```diff
@@ -515,115 +515,34 @@ func runDBCheck(opts dbCheckOptions) (err error) {
 		err = errors.Join(err, nonAncientDB.Close())
 	}()
 
-	lastAncient, err := loadLastAncient(ancientDB)
+	ancients, err := ancientDB.Ancients()
 	if err != nil {
-		return fmt.Errorf("failed to load last ancient block: %w", err)
+		return fmt.Errorf("failed to call Ancients: %w", err)
 	}
-	lastAncientNumber := lastAncient.Header().Number.Uint64()
-	lastBlockNumber := *rawdb.ReadHeaderNumber(nonAncientDB, rawdb.ReadHeadHeaderHash(nonAncientDB))
-
-	var errResult *multierror.Error
-
-	// First, check continuity between ancients and non-ancients.
-	// Gaps in data will often halt the freezing process, so attempting to load the first non-ancient block
-	// will most likely fail if there is a gap.
-	firstNonAncientRange, err := loadNonAncientRange(nonAncientDB, lastAncientNumber+1, 1)
+	// Check the range covering the ancient to non-ancient transition first, since this is where we previously encountered continuity issues.
+	err = checkRange(ancientDB, nonAncientDB, ancients-1, ancients+1)
 	if err != nil {
-		if opts.failFast {
```

Comment:
Just making a note that we should also remove this flag if we want to remove this functionality. Happy to take care of this after merging